src/states/menus/join_game_menu.gd

```diff
@@ -93,6 +93,7 @@ func populate() -> void:
 	while server_grid.get_child_count() > 1:
 		var old_row := server_grid.get_child(1)
 		server_grid.remove_child(old_row)
+		old_row.get_node("CheckBox").button_group = null
```

Comment:
This pull request is too massive - maybe this line should be moved into another PR with a smaller scope?